src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -46,91 +48,44 @@ public override void Initialize (AnalysisContext context)
 					if (call.IsVirtual && call.TargetMethod.OverriddenMethod != null)
 						return;
 
-					CheckMethodOrCtorCall (operationContext, call.TargetMethod, call.Syntax.GetLocation ());
+					CheckMethodOrCtorCall (operationContext, call.TargetMethod);
 				}, OperationKind.Invocation);
 
 				context.RegisterOperationAction (operationContext => {
 					var call = (IObjectCreationOperation) operationContext.Operation;
-					CheckMethodOrCtorCall (operationContext, call.Constructor, call.Syntax.GetLocation ());
+					CheckMethodOrCtorCall (operationContext, call.Constructor);
 				}, OperationKind.ObjectCreation);
 
 				context.RegisterOperationAction (operationContext => {
 					var propAccess = (IPropertyReferenceOperation) operationContext.Operation;
 					var prop = propAccess.Property;
 					var usageInfo = propAccess.GetValueUsageInfo (prop);
-					if (usageInfo.HasFlag (ValueUsageInfo.Read) && prop.GetMethod != null) {
-						CheckMethodOrCtorCall (
-							operationContext,
-							prop.GetMethod,
-							propAccess.Syntax.GetLocation ());
-					}
-					if (usageInfo.HasFlag (ValueUsageInfo.Write) && prop.SetMethod != null) {
-						CheckMethodOrCtorCall (
-							operationContext,
-							prop.SetMethod,
-							propAccess.Syntax.GetLocation ());
-					}
+					if (usageInfo.HasFlag (ValueUsageInfo.Read) && prop.GetMethod != null)
+						CheckMethodOrCtorCall (operationContext, prop.GetMethod);
+
+					if (usageInfo.HasFlag (ValueUsageInfo.Write) && prop.SetMethod != null)
+						CheckMethodOrCtorCall (operationContext, prop.SetMethod);
 				}, OperationKind.PropertyReference);
 
-				void CheckMethodOrCtorCall (
+				static void CheckMethodOrCtorCall (
 					OperationAnalysisContext operationContext,
-					IMethodSymbol method,
-					Location location)
+					IMethodSymbol method)
 				{
-					AttributeData? requiresUnreferencedCode;
 					// If parent method contains RequiresUnreferencedCodeAttribute then we shouldn't report diagnostics for this method
 					if (operationContext.ContainingSymbol is IMethodSymbol &&
-						TryGetRequiresUnreferencedCodeAttribute (operationContext.ContainingSymbol.GetAttributes (), out requiresUnreferencedCode))
+						operationContext.ContainingSymbol.TryGetAttributeWithMessageOnCtor (FullyQualifiedRequiresUnreferencedCodeAttribute, out AttributeData? requiresUnreferencedCode))
```

Comment:
I think we could use the same operationContext.ContainingSymbol.HasAttribute() function here